### PR TITLE
Get our test suite passing on TruffleRuby

### PR DESCRIPTION
**Pull request by ivoanjo**
The minor changes in this PR are enough to run both `bundle exec rake spec:main` and `bundle exec rake test:main` on TruffleRuby with only 2 tests failing.

The 2 failing tests are due to https://github.com/oracle/truffleruby/issues/2273 so we'll need an upstream fix for them.

Thanks to @P403n1x87 for pairing with me on this.

### CODE REVIEW

File: spec/support/platform_helpers.rb [hunk 1]
```diff
@@ -9,7 +9,11 @@ def jruby?
     RUBY_ENGINE == 'jruby'.freeze
   end
 
+  def truffleruby?
+    RUBY_ENGINE == 'truffleruby'.freeze
+  end
+
   def supports_fork?
```
Comment (eregon) [inline spec/support/platform_helpers.rb:16]: In general, a better way to check that is `Process.respond_to?(:fork)`:
https://github.com/oracle/truffleruby/blob/master/doc/user/compatibility.md#fork

**Review by marcotc — APPROVED** [review]
I'm super pleased with how few changes were necessary to get our main tests to pass.

Next steps would be to add truffleruby to our CircleCI matrix, and ensure that `rake ci` executes only the test suites we know are currently passing with truffleruby.

I think that @eregon's suggestion is a good one. With his suggestion added, I'm 👍 for merging this PR.

File: spec/support/platform_helpers.rb (hunk 1 above)
Comment (ivoanjo) [reply spec/support/platform_helpers.rb]: Fixed in 29e73f59fc910ee1113696f86aa02fd95e01fff3, thanks for the hint :)

**Comment by ivoanjo** [comment]
> Next steps would be to add truffleruby to our CircleCI matrix, and ensure that rake ci executes only the test suites we know are currently passing with truffleruby.

This! I already have a docker image ready at `ivoanjo/docker-library:ddtrace_rb_truffleruby_21_0_0` and I'm running it locally and seeing what we can/need to fix, and what needs to be pushed upstream.

By the end of the week I plan to open at least one other PR to add the `Dockerfile`/`docker-compose.yml` configuration.

**Review by delner — APPROVED** [review]
Changes look fine to me; I like that it's more direct with checking the `Process` directly rather than hiding it in `PlatformHelpers`.

**Pull request merged**